instantly/read.go

```diff
@@ -0,0 +1,102 @@
+package instantly
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/handy"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+const (
+	objectNameCampaigns = "campaigns"
+	objectNameAccounts  = "accounts"
+	objectNameEmails    = "emails"
+	objectNameTags      = "tags"
+)
+
+var supportedObjectsByRead = handy.NewSet([]string{ //nolint:gochecknoglobals
+	// Object Name	----------	API endpoint path
+	objectNameCampaigns, // campaign/list
+	objectNameAccounts,  // account/list
+	objectNameEmails,    // unibox/emails
+	objectNameTags,      // custom-tag
+})
+
+func (c *Connector) Read(ctx context.Context, config common.ReadParams) (*common.ReadResult, error) {
+	if len(config.ObjectName) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	if !supportedObjectsByRead.Has(config.ObjectName) {
+		return nil, common.ErrOperationNotSupportedForObject
+	}
+
+	urlPath, nodePath, err := matchObjectNameToEndpointPath(config.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	link, err := c.buildReadURL(config, urlPath)
+	if err != nil {
+		return nil, err
+	}
+
+	rsp, err := c.Client.Get(ctx, link.String())
+	if err != nil {
+		return nil, err
+	}
+
+	return common.ParseResult(
+		rsp,
+		makeGetTotalSize(nodePath),
+		makeGetRecords(nodePath),
+		makeNextRecordsURL(nodePath, link),
+		common.GetMarshaledData,
+		config.Fields,
+	)
+}
+
+func matchObjectNameToEndpointPath(objectName string) (urlPath string, nodePath string, err error) {
+	switch objectName {
+	case objectNameCampaigns:
+		// https://developer.instantly.ai/campaign-1/list-campaigns
+		// Empty string of data location means the response is an array itself holding what we need.
+		return "campaign/list", "", nil
+	case objectNameAccounts:
+		// https://developer.instantly.ai/account/list-accounts
+		return "account/list", "accounts", nil
+	case objectNameEmails:
+		// https://developer.instantly.ai/unibox/emails-or-list
+		return "unibox/emails", "data", nil
+	case objectNameTags:
+		// https://developer.instantly.ai/tags/list-tags
+		return "custom-tag", "data", nil
+	default:
+		return "", "", common.ErrOperationNotSupportedForObject
+	}
+}
+
+func (c *Connector) buildReadURL(config common.ReadParams, urlPath string) (*urlbuilder.URL, error) {
+	if len(config.NextPage) != 0 {
+		// Next page
+		link, err := urlbuilder.New(config.NextPage.String())
+		if err != nil {
+			return nil, err
+		}
+
+		return link, nil
+	}
+
+	// First page
+	link, err := c.getURL(urlPath)
```

Comment:
let's use `url` instead of `link`

Reply:
Sorry, missed that one.
I did rename in other files too. This naming will stop.

Reply:
It started to get messy for this PR, I split it: https://github.com/amp-labs/connectors/pull/993